Controllers/ProductsController.cs

```diff
@@ -165,24 +169,53 @@ public IActionResult New()
             ViewData["ProductTypeId"] = context.ProductType
                 .OrderBy(l => l.Label)
                 .AsEnumerable()
-                .Select(li => new SelectListItem { 
+                .Select(li => new SelectListItem
+                {
                     Text = li.Label,
                     Value = li.ProductTypeId.ToString()
-                    });
+                });
 
             ViewData["UserId"] = context.User
                 .OrderBy(l => l.LastName)
                 .AsEnumerable()
-                .Select(li => new SelectListItem { 
+                .Select(li => new SelectListItem
+                {
                     Text = $"{li.FirstName} {li.LastName}",
                     Value = li.UserId.ToString()
                 });
-            
+
             ViewBag.Users = Users.GetAllUsers(context);
 
             return View(product);
 
         }
+
+        public async Task<IActionResult> Delete([FromRoute]int id)
+        {
+            Product originalProduct = await context.Product.SingleAsync(p => p.ProductId == id);
+
+            if (!ModelState.IsValid)

```

Comment:
This shouldn't check for ModelState but instead the return of our Linq statement:
`if (originalProduct == null)`